examples/constructor/src/call.rs

```diff
@@ -246,18 +246,20 @@ mod wasm {
             Some((left == right).encode())
         }
 
-        fn if_else(self, previous: Option<CallResult>) -> Option<Vec<u8>> {
-            let Self::IfElse(flag, true_call, false_call) = self else {
+        fn if_else(self, mut previous: Option<CallResult>) -> Option<Vec<u8>> {
+            let Self::IfElse(flag, true_calls, false_calls) = self else {
                 unreachable!()
             };
 
             let flag = flag.value();
 
-            let call = if flag { true_call } else { false_call };
+            let calls = if flag { true_calls } else { false_calls };
 
-            let (_call, value) = call.process(previous);
+            for call in calls {
+                previous = Some(call.process(previous));
```

Comment:
Whats the point?
```suggestion
                previous = call.process(previous);
```

Reply:
Because the type of `previous` is Option<...>, so we just reuse it instead of making a whole new variable.